components/raft_log_engine/src/engine.rs

```diff
@@ -25,14 +29,143 @@ impl MessageExt for MessageExtTyped {
     }
 }
 
+struct ManagedReader<R: Seek + Read> {
+    raw: Option<R>,
```

Comment:
How about `tikv_util::Either` instead of `Option`?

Reply:
Good idea.